pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator.go

```diff
@@ -119,6 +120,15 @@ func (mutator *VMIsMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1
 		// Set the phase to pending to avoid blank status
 		newVMI.Status.Phase = v1.Pending
 
+		if mutator.ClusterConfig.NonRootEnabled() {
+			if canBeNonRoot(newVMI) {
+				if newVMI.ObjectMeta.Annotations == nil {
+					newVMI.ObjectMeta.Annotations = make(map[string]string)
+				}
+				newVMI.ObjectMeta.Annotations[v1.NonRootVMIAnnotation] = ""
+			}
+		}
```

Comment:
when non root is set, then `!canBeNonRoot` should cause a validation err. 

We discussed the possiblity of enforcing that both NonRoot and VirtioFS feature gates can't be enabled at the same time, however there's not a practical way of enforcing that in a validation webhook today. There's no `CREATE` validation webhook for the KubeVirt CR to help us here.

so, that said, i think non root should mean all VMIs are non root or not accepted at all. no surprises for anyone, especially when it comes to security.

Reply:
Ok, I will move it to validation webhook.